main.tf

```diff
@@ -55,6 +63,21 @@ resource "azurerm_management_lock" "this" {
   scope      = azurerm_kubernetes_cluster.this.id
 }
 
+resource "azurerm_kubernetes_cluster_node_pool" "this" {
+  # set max nodepools created to 3
+  for_each = var.node_pools
```

Comment:
You could simplify the variable node_pools. Now it is a map of objects. The variable could be just an object.

the for_each could be done on the zones available. Using the single object in the variable you create identical node pools for each zone.

Reply:
i have implemented a local now that provides the zones if they are available creates three node pools with each having an availability zone